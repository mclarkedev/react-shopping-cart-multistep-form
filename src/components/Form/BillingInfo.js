import React, { Component } from 'react';
import Button from '../../styles/elements/button';
import TwoCol from '../../styles/elements/twocol';

import Fade from 'react-reveal/Fade';

export class BillingInfo extends Component {
    continue = (e) => {
        e.preventDefault();
        this.props.nextStep();
    }
    previous = (e) => {
        e.preventDefault();
        this.props.prevStep();
    }

    render() {
        const { values, handleChange } = this.props;
        return (
            <Fade>
                <section>
                    <h2>Payment Details</h2>
                    <label htmlFor='cardholder'>
                        Name on card
                    </label>
                    <input
                        required
                        type="text"
                        name="cardholder"
                        onChange={handleChange('cardholder')}
                        defaultValue={values.cardholder}
                    />
                    <label htmlFor='creditNumber'>
                        Card number
                    </label>
                    <input
                        required
                        type="number"
                        name="creditNumber"
                        onChange={handleChange('creditNumber')}
                        defaultValue={values.creditNumber}
                    />
                    <TwoCol className="billing">
                        <div>
                            <label htmlFor='expiration'>
                                Expiration
                            </label>
                            <input
                            id="expiration"
                            placeholder='MMYY'
                            minLength="4"
                            maxLength="4"
                            type="number"
                            required
                            name="expiration"
                            onChange={handleChange('expiration')}
                            defaultValue={values.expiration}
                            />
                        </div>
                        <div>
                            <label htmlFor='cvc'>
                                CVC
                            </label>
                            <input
                                id="cvc"
                                required
                                minLength="3"
                                maxLength="3"
                                type="number"
                                name="cvc"
                                onChange={handleChange('cvc')}
                                defaultValue={values.cvc}
                            />
                        </div>
                    </TwoCol>
                    <TwoCol>
                        <Button onClick={this.previous}>Back</Button>
                        <Button 
                            disabled={!(values.cardholder, values.creditNumber, values.expiration, values.cvc)}
                            onClick={this.continue}
                        >
                            Next
                        </Button>
                    </TwoCol>
                </section>
            </Fade>
        )
    }
}

export default BillingInfo;