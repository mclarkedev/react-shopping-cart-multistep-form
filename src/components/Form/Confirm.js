import React, { Component } from 'react';

export class Confirm extends Component {
    continue = (e) => {
        e.preventDefault();
        this.props.nextStep();
    }
    previous = (e) => {
        e.preventDefault();
        this.props.prevStep();
    }

    render () {
        const {values} = this.props;
        return (
            <div>
                <h2>Confirm Order</h2>
                <p><strong>Name:</strong> {values.firstName} {values.lastName}</p>
                <p><strong>Email:</strong> {values.email}</p>
                <p><strong>Phone Number:</strong> {values.phoneNumber}</p>
                <p><strong>Shipping to:</strong>
                    <br/>
                    {values.street}
                    <br />
                    {values.city}, {values.state}
                    <br />
                    {values.zip}
                </p>
                <div className="flex-two-col">
                    <button onClick={this.previous}>Prev</button>
                    <a href="/success" className="button">Confirm &amp; Continue</a>
                </div>
        </div>
        )
    }
}

export default Confirm
