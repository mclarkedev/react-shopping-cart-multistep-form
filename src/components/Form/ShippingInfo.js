import React, { Component } from 'react'

export class ShippingInfo extends Component {
    continue = (e) => {
        e.preventDefault();
        this.props.nextStep();
    }
    previous = (e) => {
        e.preventDefault();
        this.props.prevStep();
    }

    render() {
        const { values, handleChange } = this.props;
        return (
            <section>
                <h2>Shipping Details</h2>
                <label htmlFor='street'>
                    Street Address
                </label>
                <input
                    required
                    type="text"
                    name="street"
                    onChange={handleChange('street')}
                    defaultValue={values.street}
                />
               <label htmlFor='city'>
                    City
                </label>
                <input
                    required
                    type="text"
                    name="city"
                    onChange={handleChange('city')}
                    defaultValue={values.city}
                />
                <div className="flex-two-col shipping">
                    <div>
                        <label htmlFor='state'>
                            State
                        </label>
                        <select 
                            required
                            name="state"
                            onChange={handleChange('state')}
                            defaultValue={values.state}
                        >
                            <option value="AL">AL</option>
                            <option value="AK">AK</option>
                            <option value="AZ">AZ</option>
                            <option value="AR">AR</option>
                            <option value="CA">CA</option>
                            <option value="CO">CO</option>
                            <option value="CT">CT</option>
                            <option value="DE">DE</option>
                            <option value="DC">DC</option>
                            <option value="FL">FL</option>
                            <option value="GA">GA</option>
                            <option value="HI">HI</option>
                            <option value="ID">ID</option>
                            <option value="IL">IL</option>
                            <option value="IN">IN</option>
                            <option value="IA">IA</option>
                            <option value="KS">KS</option>
                            <option value="KY">KY</option>
                            <option value="LA">LA</option>
                            <option value="ME">ME</option>
                            <option value="MD">MD</option>
                            <option value="MA">MA</option>
                            <option value="MI">MI</option>
                            <option value="MN">MN</option>
                            <option value="MS">MS</option>
                            <option value="MO">MO</option>
                            <option value="MT">MT</option>
                            <option value="NE">NE</option>
                            <option value="NV">NV</option>
                            <option value="NH">NH</option>
                            <option value="NJ">NJ</option>
                            <option value="NM">NM</option>
                            <option value="NY">NY</option>
                            <option value="NC">NC</option>
                            <option value="ND">ND</option>
                            <option value="OH">OH</option>
                            <option value="OK">OK</option>
                            <option value="OR">OR</option>
                            <option value="PA">PA</option>
                            <option value="RI">RI</option>
                            <option value="SC">SC</option>
                            <option value="SD">SD</option>
                            <option value="TN">TN</option>
                            <option value="TX">TX</option>
                            <option value="UT">UT</option>
                            <option value="VT">VT</option>
                            <option value="VA">VA</option>
                            <option value="WA">WA</option>
                            <option value="WV">WV</option>
                            <option value="WI">WI</option>
                            <option value="WY">WY</option>
                        </select>
                    </div>
                    <div>
                        <label htmlFor='zip'>
                            Zip
                        </label>
                        <input
                            required
                            type="number"
                            name="zip"
                            onChange={handleChange('zip')}
                            defaultValue={values.zip}
                        />
                    </div>
                </div>
                <div className="flex-two-col">
                    <button onClick={this.previous}>Prev</button>
                    <button onClick={this.continue}>Next</button>
                </div>
            </section>
        )
    }
}
export default ShippingInfo;