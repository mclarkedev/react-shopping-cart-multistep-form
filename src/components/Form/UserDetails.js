import React, { Component } from 'react';
import Button from '../../styles/elements/button';

import Fade from 'react-reveal/Fade';

export class UserDetails extends Component {
    continue = (e) => {
        e.preventDefault();
        this.props.nextStep();
    }

    render() {
        const { values, handleChange } = this.props;
        return (
            <Fade>
                <section>
                    <h2>Account Details</h2>
                    <label htmlFor='firstName'>
                        First Name
                    </label>
                    <input
                        required
                        type="text"
                        name="firstName"
                        onChange={handleChange('firstName')}
                        defaultValue={values.firstName}
                    />
                    <label htmlFor='lastName'>
                        Last Name
                    </label>
                    <input
                        required
                        type="text"
                        name="lastName"
                        onChange={handleChange('lastName')}
                        defaultValue={values.lastName}
                    />
                    <label htmlFor='email'>
                        Email Address
                    </label>
                    <input
                        required
                        pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$"
                        type="email"
                        name="email"
                        onChange={handleChange('email')}
                        defaultValue={values.email}
                    />
                    <label htmlFor='phoneNumber'>
                        Phone Number
                    </label>
                    <input
                        required
                        type="tel"
                        minLength="10"
                        maxLength="14"
                        placeholder="1-123-456-7890"
                        name="phoneNumber"
                        onChange={handleChange('phoneNumber')}
                        defaultValue={values.phoneNumber}
                    />
                    <Button 
                        disabled={!(values.firstName, values.lastName, values.email, values.phoneNumber)}                    
                        onClick={this.continue}
                    >
                        Next
                    </Button>
                </section>
            </Fade>
        )
    }
}

export default UserDetails;