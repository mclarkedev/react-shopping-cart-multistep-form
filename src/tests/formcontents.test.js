import React from 'react';
import { shallow } from 'enzyme';
import values from './testdata';
import { nextStep, prevStep, handleChange } from './testfunctions';
import Button from '../styles/elements/button';

/* Form Components */
import UserDetails from '../components/Form/UserDetails';
import ShippingInfo from '../components/Form/ShippingInfo';
import BillingInfo from '../components/Form/BillingInfo';
import Confirm from '../components/Form/Confirm';

describe('Form Pages', () => {

    describe('User Details', () => {
        const component = shallow(
            <UserDetails 
            nextStep={nextStep}
            handleChange={handleChange}
            values={values}/>);     
        it('matches snapshot', () => {
            expect(component).toMatchSnapshot();
        })
        it('should have one button', () => {
            const wrapper = component.find(Button);
            expect(wrapper.length).toBe(1);
        })
        it('should have four inputs', () => {
            const wrapper = component.find('input');
            expect(wrapper.length).toBe(4);
        });
    })

    describe('Shipping Info', () => {
        const component = shallow(
            <ShippingInfo 
            nextStep={nextStep}
            prevStep={prevStep}
            handleChange={handleChange}
            values={values}/>);
        it('matches snapshot', () => {
            expect(component).toMatchSnapshot();
        })
        it('should have two buttons', () => {
            const wrapper = component.find(Button);
            expect(wrapper.length).toBe(2);
        })
        it('should have three inputs', () => {
            const wrapper = component.find('input');
            expect(wrapper.length).toBe(3);
        }); 
    })

    describe('Billing Info', () => {
        const component = shallow(
            <BillingInfo
            nextStep={nextStep}
            prevStep={prevStep}
            handleChange={handleChange}
            values={values}/>);
        it('matches snapshot', () => {
            expect(component).toMatchSnapshot();
        })
        it('should have two buttons', () => {
            const wrapper = component.find(Button);
            expect(wrapper.length).toBe(2);
        })
        it('should have four inputs', () => {
            const wrapper = component.find('input');
            expect(wrapper.length).toBe(4);
        }); 
    })

    describe('Confirm', () => {
        const component = shallow(
            <Confirm
            nextStep={nextStep}
            prevStep={prevStep}
            values={values}/>);
        it('matches snapshot', () => {
            expect(component).toMatchSnapshot();
        })
        it('should have one button', () => {
            const wrapper = component.find(Button);
            expect(wrapper.length).toBe(1);
        })
        it('should have four inputs', () => {
            const wrapper = component.find('p');
            expect(wrapper.length).toBe(4);
        }); 
    });
});
