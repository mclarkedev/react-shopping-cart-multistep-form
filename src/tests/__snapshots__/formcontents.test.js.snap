// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Pages Billing Info matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Form Pages Confirm matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Form Pages Shipping Info matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Form Pages User Details matches snapshot 1`] = `ShallowWrapper {}`;
